### ISSUE TITLE
Missing neural network operations

### ISSUE THREAD

**Issue by avik-pal** [EnzymeAD/Reactant.jl#250]
## NNlib

- [x] activations
    - [x] tanh_fast
    - [x] sigmoid
    - [x] sigmoid_fast
- [x] softmax!
- [x] logsoftmax!
- [x] conv
- [ ] conv!
- [x] maxpool
- [ ] maxpool!
- [x] meanpool
- [ ] meanpool!
- [x] batched_mul
- [ ] batched_mul!
- [ ] padding
    - [x] pad_constant
    - [ ] pad_circular
    - [ ] pad_repeat
    - [x] pad_zeros
    - [ ] pad_reflect
    - [ ] pad_symmetric
- [ ] ∇conv_data (needed for ConvTranspose)
- [ ] ∇conv_data!
- [ ] ∇conv_filter
- [ ] ∇conv_filter!
- [ ] gather
- [ ] gather!
- [ ] scatter
- [ ] scatter!
- [ ] ~dot_product_attention~
- [ ] pixel_shuffle
- [ ] ~batchnorm~

Some of these might not even be needed, but we should test the generated IR. Strikethrough denotes we don't need specialized handling for these operations

Feel free to add missing operations to the list
